bgpd/bgp_packet.c

```diff
@@ -1530,11 +1531,17 @@ bgp_update_receive (struct peer *peer, bgp_size_t size)
         {
           case NLRI_UPDATE:
           case NLRI_MP_UPDATE:
-            nlri_ret = bgp_nlri_parse (peer, NLRI_ATTR_ARG, &nlris[i]);
+            if (nlris[i].afi == AFI_L2VPN && nlris[i].safi == SAFI_EVPN)
+              nlri_ret = bgp_nlri_parse_evpn (peer, NLRI_ATTR_ARG, &nlris[i], 0);
+            else
+              nlri_ret = bgp_nlri_parse (peer, NLRI_ATTR_ARG, &nlris[i]);
             break;
           case NLRI_WITHDRAW:
           case NLRI_MP_WITHDRAW:
-            nlri_ret = bgp_nlri_parse (peer, NULL, &nlris[i]);
+            if (nlris[i].afi == AFI_L2VPN && nlris[i].safi == SAFI_EVPN)
```

Comment:
I will have to extend bgp_nlri_parse with one extra parameter.
reason is that i need to pass attr structure, even for mp_unreach messages.

yes, i can do the change.